### PR TITLE
Update _ncomp.pyx and rcm-related test scripts

### PR DESCRIPTION
**Pull request by khallock**
Files were copied from Issue_60 branch on NCAR/geocat-comp@46df7be (_ncomp.pyx) and NCAR/geocat-comp@10acbc2 (test scripts).

This commit implements some of the changes proposed in PR NCAR/geocat-comp#64.

### CODE REVIEW

**Comment by khallock** [comment]
I reverted the merge of PR #1 so that we can go through the code review process, but it isn't possible to reopen a PR like it is for an issue so I had to create a new PR.

I forgot to enable the PR merge restrictions on this repo that require checks and reviews to be completed prior to merging, sorry for the confusion @pilotchute! Let me know if this throws your local repo out of sync, I'd be glad to help out!

**Comment by pilotchute** [comment]
Sounds good. My local repo will be fine. 

What do I need to do to perform an approving review?

**Comment by khallock** [comment]
There should be a "view changes" button somewhere on this page (I'm not 100% sure what the interface looks like, I can't add myself as a reviewer), if you click that it should bring you to a page with all of the file diffs. At the top of that page, there should be another green "review changes" button that lets you add comments and choose a status from a list of "approve", "comment", and "request changes". Thanks!

File: test/test_rcm2points.py [hunk 1]
```diff
@@ -1,75 +1,87 @@
 import numpy as np
+import numpy.testing as nt
 import xarray as xr
-import geocat.ncomp
+import geocat.ncomp as gn
```
Comment (pilotchute) [inline test/test_rcm2points.py:4]: Should this alias become standard? Or is it unneeded?

File: test/test_rcm2points.py [hunk 2]
```diff
@@ -1,75 +1,87 @@
...
+
+# expected output
+fo_nom_opt0_expected = np.asarray([1.870327, 1.353965, 1.588746, -0.1676207, 1.01385, 0.7974159, 2.015617, 0.6676366, 1.249507]).reshape((3, 3))
+fo_nom_opt2_expected = np.asarray([1.870327, 1.353965, 1.588746, -0.1676207, 1.01385, 0.7974159, 2.015617, 0.6676366, 1.249507]).reshape((3, 3))
+
+fo_nan_opt0_expected = np.asarray([1.870327, 1.486811, 1.679019, -0.1676207, 0.9685476, 0.7141976, 2.015617, 1.757489, 1.473235]).reshape((3, 3))
+fo_nan_opt2_expected = np.asarray([1.870327, 1.851139, 1.679019, -0.1676207, 0.2114156, 0.7141976, 2.015617, 0.9372569, 1.473235]).reshape((3, 3))
+
+fo_msg_opt0_expected = np.asarray([1.870327, 1.486811, 1.679019, -0.1676207, 0.9685476, 0.7141976, 2.015617, 1.757489, 1.473235]).reshape((3, 3))
+fo_msg_opt2_expected = np.asarray([1.870327, 1.851139, 1.679019, -0.1676207, 0.2114156, 0.7141976, 2.015617, 0.9372569, 1.473235]).reshape((3, 3))
```
Comment (pilotchute) [inline test/test_rcm2points.py:43]: This input/output comes from a new .ncl file, should that file be added to this commit?

File: test/test_rcm2rgrid.py [hunk 3]
```diff
@@ -1,69 +1,71 @@
...
+    [1.870327, 1.98253, 0.1410671, 1.872924, 1.353965, 1.877125, 2.946794, 0.8730035, 1.931963, -0.1676207, -1.82497, 1.754721, 1.917912, 1.01385, 1.177423,
+     1.735453, 1.053591, 0.381366, 2.015617, 0.3293635, 2.510986, 0.4975608, 0.6676366, 1.027274, 2.169137, 2.691788, 1.351906]).reshape((3, 3, 3))
+
+fo_nan_expected = np.array(
+    [1.870327, 1.98253, 0.1410671, 1.872924, 1.875024, 1.877125, 2.946794, 0.8730035, 1.931963, -0.1676207, -1.82497, 1.754721, 1.917912, 1.547667, 1.177423,
+     1.735453, 1.053591, 0.381366, 2.015617, 0.3293635, 2.510986, 0.4975608, 0.7624173, 1.027274, 2.169137, 2.691788, 1.351906]).reshape((3, 3, 3))
+
+fo_msg_expected = np.array(
+    [1.870327, 1.98253, 0.1410671, 1.872924, 1.875024, 1.877125, 2.946794, 0.8730035, 1.931963, -0.1676207, -1.82497, 1.754721, 1.917912, 1.547667, 1.177423,
+     1.735453, 1.053591, 0.381366, 2.015617, 0.3293635, 2.510986, 0.4975608, 0.7624173, 1.027274, 2.169137, 2.691788, 1.351906]).reshape((3, 3, 3))
```
Comment (pilotchute) [inline test/test_rcm2rgrid.py:45]: This input/output comes from a new .ncl file, should that file be added to this commit?

File: test/test_rgrid2rcm.py [hunk 4]
```diff
@@ -1,69 +1,70 @@
...
+    [1.870327, 1.98253, 0.1410671, 1.872924, 1.353965, 1.877125, 2.946794, 0.8730035, 1.931963, -0.1676207, -1.82497, 1.754721, 1.917912, 1.01385, 1.177423,
+     1.735453, 1.053591, 0.381366, 2.015617, 0.3293635, 2.510986, 0.4975608, 0.6676366, 1.027274, 2.169137, 2.691788, 1.351906]).reshape((3, 3, 3))
+
+fo_nan_expected = np.array(
+    [1.870327, 1.98253, 0.1410671, 1.872924, 1.486321, 1.877125, 2.946794, 0.8730035, 1.931963, -0.1676207, -1.82497, 1.754721, 1.917912, 0.9684872, 1.177423,
+     1.735453, 1.053591, 0.381366, 2.015617, 0.3293635, 2.510986, 0.4975608, 1.7583, 1.027274, 2.169137, 2.691788, 1.351906]).reshape((3, 3, 3))
+
+fo_msg_expected = np.array(
+    [1.870327, 1.98253, 0.1410671, 1.872924, 1.486321, 1.877125, 2.946794, 0.8730035, 1.931963, -0.1676207, -1.82497, 1.754721, 1.917912, 0.9684872, 1.177423,
+     1.735453, 1.053591, 0.381366, 2.015617, 0.3293635, 2.510986, 0.4975608, 1.7583, 1.027274, 2.169137, 2.691788, 1.351906]).reshape((3, 3, 3))
```
Comment (pilotchute) [inline test/test_rgrid2rcm.py:44]: This input/output comes from a new .ncl file, should that file be added to this commit?

File: test/test_rcm2points.py (hunk 1 above)
Comment (khallock) [reply test/test_rcm2points.py:4]: I think this is generally unneeded, I was just translating `import geocat.comp as gc` from the test scripts that had been in the PR on GeoCAT-comp.

File: test/test_rcm2points.py (hunk 2 above)
Comment (khallock) [reply test/test_rcm2points.py:43]: Per our meeting today (and NCAR/ncomp#25), I think example NCL scripts only need to exist in the NComp repo.

File: test/test_rcm2rgrid.py (hunk 3 above)
Comment (khallock) [reply test/test_rcm2rgrid.py:45]: See above

File: test/test_rgrid2rcm.py (hunk 4 above)
Comment (khallock) [reply test/test_rgrid2rcm.py:44]: See above

**Pull request merged**